src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -259,111 +261,96 @@ public static void main(String... args) throws IOException {
     }
 
     /**
-     * Parses and executes Checkstyle based on passed arguments.
-     * @param args
-     *        command line parameters
-     * @return parsed information about passed parameters
-     * @throws ParseException
-     *         when passed arguments are not valid
+     * Validates the user input and returns {@value EXIT_WITH_INVALID_USER_INPUT_CODE} if
+     * invalid, otherwise executes CheckStyle and returns the number of violations.
+     * @return number of violations
+     * @throws IOException if a file could not be read.
+     * @throws CheckstyleException if something happens processing the files.
      */
-    private static CommandLine parseCli(String... args)
-            throws ParseException {
-        // parse the parameters
-        final CommandLineParser clp = new DefaultParser();
-        // always returns not null value
-        return clp.parse(buildOptions(), args);
+    @Override
+    public Integer call() throws IOException, CheckstyleException {
```

Comment:
@romani This is an overridden method, so it can't be renamed. The class implements `java.util.concurrent.Callable` which is forcing the `call` method name.

`Integer` is because `Callable` has a template and I assume it accept primitive types.

Reply:
Will do.

Reply:
Done